File: personal_project_back-end/main-service/src/main/java/com/example/mainservice/models/InvoiceServiceInfoDetailModel.java
package com.example.mainservice.models;

import com.example.common.service.enums.InvoiceStatusEnum;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Data
public class InvoiceServiceInfoDetailModel {
    private Long invoiceId;
    private LocalDateTime invoiceCreated;
    private int monthCal;
    private int yearCal;
    private BigDecimal totalInvoice;
    private String fullName;
    private Long roomId;
    private BigDecimal roomPrice;
    private InvoiceStatusEnum status;
    private List<ServiceInfoModel> serviceInfos;
}

File: personal_project_front-end/src/models/InvoiceServiceInfoDetailModel.ts
import { InvoiceStatusEnum } from "@/enums/InvoiceStatusEnum";

export type ServiceInfoModel = {
  serviceId: number;
  nameService: string;
  serviceType: string;
  priceService: number;
  quantityService: number;
};

export type InvoiceServiceInfoDetailModel = {
  invoiceId: number;
  invoiceCreated: string;
  monthCal: number;
  yearCal: number;
  totalInvoice: number;
  fullName: string;
  roomId: number;
  status: InvoiceStatusEnum;
  serviceInfos: ServiceInfoModel[];
  roomPrice: number;
};

export const defaultInvoiceServiceInfoDetailModel: InvoiceServiceInfoDetailModel =
  {
    invoiceId: 0,
    invoiceCreated: new Date().toISOString(),
    monthCal: new Date().getMonth() + 1,
    yearCal: new Date().getFullYear(),
    totalInvoice: 0,
    fullName: "",
    roomId: 0,
    status: InvoiceStatusEnum.UNPAID,
    serviceInfos: [],
    roomPrice: 0,
  };
